src/luarocks/manif.lua

```diff
@@ -344,6 +343,8 @@ local function store_results(results, manifest)
    assert(type(results) == "table")
    assert(type(manifest) == "table")
 
+   local repos = require('luarocks.repos')
```

Comment:
I would comment it like 'prevent cyclic dependencies' just to give a reader an idea that this has the certain reason.